bepasty_cli/cli.py

```diff
@@ -18,7 +19,35 @@
 from requests.packages.urllib3.exceptions import InsecureRequestWarning
 
 
-@click.command()
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+LIFETIME_CHOICES = (
+    'min', 'minutes', 'h', 'hours', 'd', 'days', 'w', 'weeks', 'm', 'months',
+    'y', 'years', 'f', 'forever'
+)
+LIFETIME_NAMES = (
+    'MINUTES', 'MINUTES', 'HOURS', 'HOURS', 'DAYS', 'DAYS', 'WEEKS', 'WEEKS',
+    'MONTHS', 'MONTHS', 'YEARS', 'YEARS', 'FOREVER', 'FOREVER'
+)
+LIFETIME_MAPPING = dict(zip(LIFETIME_CHOICES, LIFETIME_NAMES))
+
+
+class LifetimeParamType(click.ParamType):
+    name = 'lifetime'
+    lifetime_regex = re.compile(r'|'.join(r'^(\d+?) ?({})$'.format(choice) for choice in LIFETIME_CHOICES))
+
+    def convert(self, value, param, ctx):
+        m = self.lifetime_regex.match(value)
+        if m:
+            result = [x for x in m.groups() if x is not None]
+            result[0] = result[0].lstrip('0')
+            if not result[0]:
+                self.fail('Multiplier for the lifetime argument must be positive.', param, ctx)
```

Comment:
hmm, sorry, I was somehow thinking about converting to int and doing int-based processing, but didn't consider we work with strings only here.

so, guess your initial version was better than this alternative.